monkey/infection_monkey/post_breach/pba.py

```diff
@@ -12,57 +15,69 @@ class PBA(object):
+        :param command: Command that will be executed on breached machine
         """
-        self.linux_command = linux_command
-        self.windows_command = windows_command
+        self.command = command
         self.name = name
 
-    def run(self, is_linux):
+    @staticmethod
+    def get_pba():
         """
-        Runs post breach action command
-        :param is_linux: boolean that indicates on which os monkey is running
+        Should be overridden by all child classes.
+        This method returns a PBA object based on worm's configuration.
+        :return: An array of PBA objects.
         """
-        if is_linux:
-            command = self.linux_command
-            exec_funct = self._execute_linux
-        else:
-            command = self.windows_command
-            exec_funct = self._execute_win
-        if command:
-            hostname = socket.gethostname()
-            ControlClient.send_telemetry('post_breach', {'command': command,
-                                                         'result': exec_funct(),
-                                                         'name': self.name,
-                                                         'hostname': hostname,
-                                                         'ip': socket.gethostbyname(hostname)
-                                                         })
+        raise NotImplementedError()
 
-    def _execute_linux(self):
+    @staticmethod
+    def default_get_pba(name, pba_class, linux_cmd="", windows_cmd=""):
         """
-        Default linux PBA execution function. Override it if additional functionality is needed
+        Default get_pba() method implementation
+        :param name: PBA name
+        :param pba_class: class instance. Class's name is matched to config to determine
+        if corresponding field was enabled in post breach array or not.
+        :param linux_cmd: commands for linux
+        :param windows_cmd: commands for windows
+        :return: post breach action
         """
-        return self._execute_default(self.linux_command)
+        if pba_class.__name__ in WormConfiguration.post_breach_actions:
+            command = PBA.choose_command(linux_cmd, windows_cmd)
+            if command:
+                return PBA(name, command)
 
-    def _execute_win(self):
+    def run(self):
         """
-        Default linux PBA execution function. Override it if additional functionality is needed
+        Runs post breach action command
         """
-        return self._execute_default(self.windows_command)
+        exec_funct = self._execute_default
+        hostname = socket.gethostname()
+        ControlClient.send_telemetry('post_breach', {'command': self.command,
+                                                     'result': exec_funct(),
+                                                     'name': self.name,
+                                                     'hostname': hostname,
```

Comment:
Why do we need hostname here, clearly the monkey has already sent the hostname as part of `wakeup` in `control.py`.